rocketpy/Function.py

```diff
@@ -2116,79 +2227,96 @@ def __sub__(self, other):
             if (
                 isinstance(other.source, np.ndarray)
                 and isinstance(self.source, np.ndarray)
-                and self.__interpolation__ == other.__interpolation__
-                and self.__inputs__ == other.__inputs__
+                and self.__domDim__ == other.__domDim__
+                and np.any(self.xArray - other.xArray) == False
```

Comment:
Is this line really needed? It seems redundant with the line below that checks that the arrays are equivalent.

Reply:
Not needed.